test/k8sT/Policies.go

```diff
@@ -535,13 +545,20 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
 			webPods, err := kubectl.GetPodsNodes(helpers.DefaultNamespace, "-l k8s-app.guestbook=web")
 			Expect(err).To(BeNil(), "Cannot get web pods")
 
+			serviceIP, port, err := kubectl.GetServiceHostPort(helpers.DefaultNamespace, "redis-master")
+
 			for pod := range webPods {
-				command := `nc redis-master 6379 <<EOF
-							PING
-							EOF`
-				res := kubectl.ExecPodCmd(helpers.DefaultNamespace, pod, command)
-				ExpectWithOffset(1, res.WasSuccessful()).To(BeTrue(),
-					"Web pod %q cannot connect to redis", pod)
+
+				// Access both service IP / host name of redis-master.
+				redisMetadata := map[string]int{serviceIP: port, "redis-master": port}
+				for k, v := range redisMetadata {
+					command := fmt.Sprintf(`nc %s %d <<EOF
+PING
+EOF`, k, v)
+					res := kubectl.ExecPodCmd(helpers.DefaultNamespace, pod, command)
+					ExpectWithOffset(1, res.WasSuccessful()).To(BeTrue(),
+						"Web pod %q cannot connect to redis", pod)
```

Comment:
I would use the following assert message: 
`%q cannot connect to redis on '%s:%i'`, pod, k, v)

If not it's not clear what trying to connect.